src/coreclr/jit/lowerxarch.cpp

```diff
@@ -9137,431 +9137,226 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* parentNode, GenTre
-                case NI_GFNI_V256_GaloisFieldAffineTransformInverse:
-                case NI_GFNI_V512_GaloisFieldAffineTransform:
-                case NI_GFNI_V512_GaloisFieldAffineTransformInverse:
-                {
-                    assert(!supportsSIMDScalarLoads);
-
-                    const unsigned expectedSize = genTypeSize(parentNode->GetSimdBaseType());
-                    const unsigned operandSize  = genTypeSize(childNode->TypeGet());
-
-                    supportsAlignedSIMDLoads = !comp->canUseVexEncoding() || !comp->opts.MinOpts();
-                    supportsGeneralLoads     = comp->canUseVexEncoding() && (operandSize >= expectedSize);
-                    break;
-                }
-
-                case NI_SSE2_ShiftLeftLogical128BitLane:
-                case NI_SSE2_ShiftRightLogical128BitLane:
-                case NI_AVX2_ShiftLeftLogical128BitLane:
-                case NI_AVX2_ShiftRightLogical128BitLane:
-                case NI_AVX512BW_ShiftLeftLogical128BitLane:
-                case NI_AVX512BW_ShiftRightLogical128BitLane:
                 {
-                    if (comp->IsBaselineVector512IsaSupportedOpportunistically())
-                    {
-                        assert(!supportsSIMDScalarLoads);
+                    assert((tupleType & INS_TT_MEM128) != 0);
 
-                        const unsigned expectedSize = genTypeSize(parentNode->GetSimdBaseType());
-                        const unsigned operandSize  = genTypeSize(childNode->TypeGet());
+                    // Shift amount (op2) can be either imm8 or vector. If vector, it will always be xmm/m128.
+                    //
+                    // Otherwise, when using EVEX encoding, we can contain a load for the input vector (op1),
+                    // so size comes from the parent.
 
-                        supportsAlignedSIMDLoads = !comp->canUseVexEncoding() || !comp->opts.MinOpts();
-                        supportsGeneralLoads     = comp->canUseVexEncoding() && (operandSize >= expectedSize);
+                    if (!HWIntrinsicInfo::isImmOp(parentIntrinsicId, parentNode->Op(2)))
+                    {
+                        expectedSize = genTypeSize(TYP_SIMD16);
+                        break;
                     }
-                    else
+                    else if ((expectedSize < genTypeSize(TYP_SIMD64)) && (ins != INS_vpsraq))
                     {
-                        assert(supportsAlignedSIMDLoads == false);
-                        assert(supportsGeneralLoads == false);
-                        assert(supportsSIMDScalarLoads == false);
+                        // TODO-XArch-CQ: This should really only be checking EVEX capability, however
+                        // emitter::TakesEvexPrefix doesn't currently handle requiring EVEX based on presence
+                        // of an immediate operand. For now we disable containment of op1 unless EVEX is
+                        // required for some other reason.
```

Comment:
If this can be handled easily, I'm happy to do it now. I couldn't see anything on `instrDesc` that gets me to the possible immediate operand.